internal/internal_workflow_client.go

```diff
@@ -1617,27 +1605,35 @@ func (w *workflowClientInterceptor) ExecuteWorkflow(
 		startRequest.WorkflowStartDelay = durationpb.New(in.Options.StartDelay)
 	}
 
-	var response *workflowservice.StartWorkflowExecutionResponse
-
 	grpcCtx, cancel := newGRPCContext(ctx, grpcMetricsHandler(
 		w.client.metricsHandler.WithTags(metrics.RPCTags(in.WorkflowType, metrics.NoneTagValue, in.Options.TaskQueue))),
 		defaultGrpcRetryParameters(ctx))
 	defer cancel()
 
-	response, err = w.client.workflowService.StartWorkflowExecution(grpcCtx, startRequest)
-	eagerWorkflowTask := response.GetEagerWorkflowTask()
-	if eagerWorkflowTask != nil && eagerExecutor != nil {
-		eagerExecutor.handleResponse(eagerWorkflowTask)
-	} else if eagerExecutor != nil {
-		eagerExecutor.release()
-	}
-	// Allow already-started error
 	var runID string
-	if e, ok := err.(*serviceerror.WorkflowExecutionAlreadyStarted); ok && !in.Options.WorkflowExecutionErrorWhenAlreadyStarted {
-		runID = e.RunId
-	} else if err != nil {
-		return nil, err
+	if in.Options.WorkflowOperation == nil {
+		response, err := w.client.workflowService.StartWorkflowExecution(grpcCtx, startRequest)
+
+		eagerWorkflowTask := response.GetEagerWorkflowTask()
+		if eagerWorkflowTask != nil && eagerExecutor != nil {
+			eagerExecutor.handleResponse(eagerWorkflowTask)
+		} else if eagerExecutor != nil {
+			eagerExecutor.release()
+		}
+
+		// Allow already-started error
+		if e, ok := err.(*serviceerror.WorkflowExecutionAlreadyStarted); ok && !in.Options.WorkflowExecutionErrorWhenAlreadyStarted {
+			runID = e.RunId
+		} else if err != nil {
+			return nil, err
+		} else {
+			runID = response.RunId
+		}
```

Comment:
`WorkflowExecutionAlreadyStarted` doesn't work for Update-with-Start as it is a client-side setting the Server doesn't know about, so the request will fail regardless of this setting. I've considered some alternatives to ignoring it (e.g. enforcing it to be in-sync with the `WorkflowIdConflictPolicy`), but this seems to be the best approach.

There is a case to be made to deprecate `WorkflowExecutionErrorWhenAlreadyStarted` once the `WorkflowIdConflictPolicy` is officially released.

Reply:
IMO we should make it clear on the operations field that use of the field changes the default already-started behavior of the SDK (today the default behavior is to not fail if already started, but setting this field means it will fail, correct?)

Reply:
Yes, setting the `WorkflowExecutionErrorWhenAlreadyStarted` to true will make it fail. I added a comment to that field, but I agree, adding it to the new field also is helpful 👍

Reply:
Just saying it out loud: I assume there's no reason to go all in with the default and set the `WorkflowIdConflictPolicy` to `UseExisting` for Update-with-Start to match today's default of ExecuteWorkflow? (in that world, we'd need to error on certain invalid combinations of `WorkflowExecutionErrorWhenAlreadyStarted` and `WorkflowIdConflictPolicy`)

Reply:
To clarify, if default behavior _without_ setting `WorkflowOperation` is one thing by default on already-exists and default behavior _with_ setting `WorkflowOperation` is another thing by default, we need to document on the field the difference in default behavior.